File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int cursum = 0;
        int maxsum = Integer.MIN_VALUE;
        
        for(int i = 0 ; i  < nums.length;i++){
            cursum = Math.max(nums[i] ,cursum + nums[i]);
            maxsum = Math.max(cursum,maxsum);
           
            
        }
        return maxsum;
    }
}

File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = INT_MIN;
        int n = nums.size();
        int cursum =0;
        for(int i=0;i<n;i++){
            cursum = max(nums[i],cursum + nums[i]);
            ans = max(ans,cursum);
        }
        return ans;
    }
};